client/my-sites/themes/survey/index.tsx

```diff
@@ -0,0 +1,77 @@
+import { translate } from 'i18n-calypso';
+import React from 'react';
+import Banner from 'calypso/components/banner';
+import './survey.scss';
+
+export enum SurveyType {
+	DECEMBER_2023 = 'DECEMBER_2023',
+}
+
+const surveys = new Map< SurveyType, { eventName: string; eventUrl: string } >( [
+	[
+		SurveyType.DECEMBER_2023,
+		{
+			eventName: 'theme-showcase-december-2023',
+			eventUrl: 'https://automattic.survey.fm/lits-survey-v1',
+		},
+	],
+] );
+
+type SurveyProps = {
+	survey: SurveyType;
+	condition: () => boolean;
+	title?: string | null;
+};
+
+/**
+ * We lose the state of whether the survey has been dismissed when navigating between pages.
+ * We also lose the state when navigating between the theme showcase and the theme details page because of all the re-renders and unmounts.
+ * This is a temporary solution to prevent the survey from showing up again after it's been dismissed.
+ */
+let isVisible = false;
+let isGloballyDismissed = false;
+
+const Survey = ( { survey, condition, title = null }: SurveyProps ) => {
+	const surveyData = surveys.get( survey );
+	const [ isDismissed, setIsDismissed ] = React.useState( isGloballyDismissed );
+
+	if ( ! isVisible && condition() ) {
+		isVisible = true;
+	}
+
+	if ( isDismissed || ! isVisible || ! surveyData ) {
+		return null;
+	}
+
+	const defaultTitle = translate(
+		'We’d love to hear your thoughts. Fill out this {{a}}quick survey{{/a}} on your theme selection experience.',
+		{
+			components: {
+				a: <a href={ surveyData.eventUrl } target="blank" />,
+			},
+		}
+	);
+
+	return (
+		<Banner
+			className="theme-showcase__survey"
+			title={ title || defaultTitle }
+			event={ surveyData.eventName }
+			onDismiss={ ( e: Event ) => {
+				e.stopPropagation();
+				setIsDismissed( true );
+				isGloballyDismissed = true;
+			} }
+			dismissWithoutSavingPreference={ true }
+			dismissTemporary={ true }
+			disableHref
+			showIcon={ false }
+			showLinkIcon={ false }
+			tracksImpressionName="calypso_theme_showcase_survey_impression"
+			tracksClickName="calypso_theme_showcase_survey_click"
+			tracksDismissName="calypso_theme_showcase_survey_dismiss"
```

Comment:
I have thoughts about the event handling of this banner.


1. `_impression` triggers too early the first time.
   - When navigating from the Showcase to the Details page and back, it triggers a first time before entering Details, and then a second time when navigating back to Showcase.
   - Problem: the event would overreport.
2. `_click` triggers when clicking anywhere on the banner, not just on the link.
   - Problem: the event would report incorrectly.

I don't have an obvious solution for `_impression`, as it's part of how we handle the visibility.

For `_click`, what if we added an `onClick` to the survey's link, and recorded that instead?

Reply:
Yeah, the re-render loop is messed up on the Theme Showcase and sub-components. We could just remove the events. The goal of these changes is not to get events but to get people to fill in the survey. I don't think we should block the PR because of the events since we don't get any business value from them.